### PR TITLE
Updating RSQLJPASupport to pass join hints when sorting

### PR DESCRIPTION
**Pull request by maxbettercloud**
Also, fixing a bug when join hints weren't passed

### CODE REVIEW

File: rsql-jpa/src/main/java/io/github/perplexhub/rsql/SortUtils.java [hunk 1]
```diff
@@ -1,48 +1,56 @@
...
 
     @SuppressWarnings("unchecked")
-    private static Order sortToJpaOrder(final String[] parts, final Map<String, String> propertyMapper, final Root<?> root, final CriteriaBuilder cb) {
+    private static Order sortToJpaOrder(final String[] parts, final SortSupport sortSupport, final Root<?> root, final CriteriaBuilder cb) {
         final String property = parts[0];
         final String direction = parts.length > 1 ? parts[1] : "asc";
 
-        final RSQLJPAPredicateConverter rsqljpaPredicateConverter = new RSQLJPAPredicateConverter(cb, propertyMapper);
+        final RSQLJPAPredicateConverter rsqljpaPredicateConverter =
+                new RSQLJPAPredicateConverter(cb, sortSupport.getPropertyPathMapper(), null, sortSupport.getJoinHints());
```
Comment (maxbettercloud) [inline rsql-jpa/src/main/java/io/github/perplexhub/rsql/SortUtils.java:53]: Passing join hints here

**Comment by maxbettercloud** [comment]
I can create a PR for `master` branch once #104 is merged.

**Comment by perplexhub** [comment]
> I can create a PR for `master` branch once #104 is merged.

Could you please create the PR for master branch? Thanks.

**Comment by perplexhub** [comment]
Merged, thanks.

**Pull request merged**